extensions/import/src/wizard/pages/fileConfigPage.ts

```diff
@@ -270,7 +270,14 @@ export class FileConfigPage extends ImportPage {
 		let connectionUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
 		let queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(this.model.server.providerName, azdata.DataProviderType.QueryProvider);
 
-		let query = `SELECT name FROM sys.schemas`;
+		let query = '';
+		if (this.databaseDropdown.value !== null)
+		{
+			query = `SELECT name FROM [${this.databaseDropdown.value}].sys.schemas`;
```

Comment:
So what I'm talking about is that you need to escape ]'s in the DB name by doubling them. This is because currently if you try to run that query against a DB that has a ] in its name then you'd end up with a query like this (assuming you have a DB named `my]db`)

`SELECT name FROM [my]db].sys.schemas`

which would break because it would see the ] as the end of the object name and thus have invalid syntax. 

So the fix is to do something like this 

const query = this.databaseDropdown.value ? 'SELECT name FROM [${this.databaseDropdown.value.replace(/]/g, ']]')}].sys.schemas' : 'SELECT name FROM sys.schemas';

So that you end up with a query like this :

`SELECT name FROM [my]]db].sys.schemas`

which will then run correctly.

(This also safeguards against the SQL injection issues that were mentioned in the previous PR)

Reply:
I am more concerned about spaces in databases names. If they have a bracket in their name I think they have more issues to resolve. I also don't see how SQL injection would play since this is just populating a drop down which is then selected. Unless a user has the ability to adjust the drop down I'm not seeing how they would be able to inject anything.

Reply:
Bracket-quoting the name is correct, I agree.

But we do require all queries that use values outside of our control to be correctly escaped. And database names fall under that bucket of "out of our control". Unless that is being done correctly we can't take this fix.

Reply:
I'm not quite sure how you would escape this, it's not like a quote character. The database names are coming directly from the database connection above which is in control, thus this would also be in control.

Reply:
I gave you the code above to escape it (you escape them by replacing every ] with ]]) :

```const query = this.databaseDropdown.value ? 'SELECT name FROM [${this.databaseDropdown.value.replace(/]/g, ']]')}].sys.schemas' : 'SELECT name FROM sys.schemas';```

Here's a possible situation that we are preventing with this. Some bad actor has somehow gotten access to create DBs on the server but not anything else. They create a DB with this name

`master].sys.tables; CREATE LOGIN foo --`

Then another person comes along with more permissions - such as being able to create a login. They open the dialog and your code above will create this query :

`SELECT name FROM [master].sys.tables; CREATE LOGIN foo WITH PASSWORD='12345' --].sys.schemas`

(Note the rest of the query is commented out because of the --)

This query is ran and now this bad actor has silently caused this other user to create a login on the DB. But you can run any query. This is extremely dangerous and as said - while unlikely to actually happen - is something we can't allow to be a possibility.